pg_shard--1.0--1.1.sql

```diff
@@ -12,7 +12,16 @@ AS $sync_table_metadata_to_citus$
 		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
 		dummy_shard_length CONSTANT bigint := 0;
 	BEGIN
-		-- copy shard placement metadata
+		-- grab lock for upsert
+		LOCK TABLE pg_dist_shard_placement IN EXCLUSIVE MODE;
+
+		-- perform update of shard health
+		UPDATE pg_dist_shard_placement
```

Comment:
Good catch! I had copy-pasted the `JOIN` clause onto most of the statements but eventually caught that the placements should have an equality clause on the triple you mention, not just the shard identifier.

I fixed it in the `INSERT` but you're right that I missed the `UPDATE`. Fixing (and I'll add a unit test that'll fail without this).